packages/synapse-interface/components/Portfolio/Portfolio.tsx

```diff
@@ -52,76 +51,34 @@ export const Portfolio = () => {
   const filteredPortfolioDataForBalances: NetworkTokenBalances =
     filterPortfolioBalancesWithBalances(portfolioData)
 
-  const searchInputActive: boolean = useMemo(() => {
-    return searchInput.length > 0
-  }, [searchInput])
-
-  const searchInputIsAddress: boolean = useMemo(() => {
-    return isValidAddress(searchInput)
-  }, [searchInput])
-
-  const masqueradeActive: boolean = useMemo(() => {
-    return Object.keys(searchedBalances).length > 0
-  }, [searchedBalances])
+  const {
+    isSearchInputActive,
+    isSearchInputAddress,
+    isMasqueradeActive,
+    masqueradeAddress,
+  } = useSearchInputState()
 
   const filteredSearchedPortfolioDataForBalances = useMemo(() => {
-    if (masqueradeActive) {
-      const queriedAddress: Address = Object.keys(
-        searchedBalances
-      )[0] as Address
+    if (isMasqueradeActive) {
       return {
         balances: filterPortfolioBalancesWithBalances(
-          searchedBalances[queriedAddress]
+          searchedBalances[masqueradeAddress]
         ),
-        address: queriedAddress,
+        address: masqueradeAddress,
       }
     }
     return {
       balances: {},
       address: '',
     }
-  }, [searchedBalances, masqueradeActive, searchInput])
-
-  const flattenedPortfolioData: TokenAndBalance[] = useMemo(() => {
-    const flattened: TokenAndBalance[] = []
-    const portfolio: NetworkTokenBalances = masqueradeActive
-      ? filteredSearchedPortfolioDataForBalances.balances
-      : filteredPortfolioDataForBalances
-    Object.entries(portfolio).forEach(([chainId, tokens]) => {
-      tokens.forEach((token: TokenAndBalance) => {
-        flattened.push({ ...token })
-      })
-    })
-    return flattened
-  }, [
-    masqueradeActive,
-    filteredPortfolioDataForBalances,
-    filteredSearchedPortfolioDataForBalances,
-  ])
+  }, [searchedBalances, isMasqueradeActive, searchInput])
 
-  const filteredBySearchInput: NetworkTokenBalances = useMemo(() => {
-    const searchFiltered: NetworkTokenBalances = {}
-    const fuseOptions = {
-      includeScore: true,
-      threshold: 0.33,
-      distance: 20,
-      keys: ['queriedChain.name', 'token.name', 'token.symbol'],
-    }
-    const fuse = new Fuse(flattenedPortfolioData, fuseOptions)
+  const flattenedPortfolioData = flattenData(portfolioData)
 
-    if (searchInput.length > 0) {
-      const results = fuse
-        .search(searchInput)
-        .map((i: Fuse.FuseResult<TokenAndBalance>) => i.item)
-        .forEach((item: TokenAndBalance) => {
-          const chainId: number = item.queriedChain.id
-          searchFiltered[chainId] = searchFiltered[chainId]
-            ? [...searchFiltered[chainId], item]
-            : [item]
-        })
-    }
-    return searchFiltered
-  }, [searchInput, flattenedPortfolioData])
+  const filteredBySearchInput = filterBySearchInput(
```

Comment:
Seeing this if searching for a token user doesn't hold

<img width="574" alt="Screen Shot 2024-01-22 at 10 04 35 PM" src="https://github.com/synapsecns/sanguine/assets/104046418/d54d5b73-2469-440c-85d4-3dcdb4b8a331">

<img width="570" alt="Screen Shot 2024-01-22 at 10 05 40 PM" src="https://github.com/synapsecns/sanguine/assets/104046418/760c87f4-c2d5-41e3-8352-38c2424f300d">

Reply:
Updated in https://github.com/synapsecns/sanguine/pull/1898/commits/069c599c0eaa1b7bcf14f4490fd8fa44aea0bba0

Wasn't displaying search results using filtered portfolio data, was using raw portfolio data. Thanks for catching this!

Reply:
Disconnected, masq mode, valid address, search comes up with no results (vs when connected and masq).

<img width="602" alt="Screen Shot 2024-01-23 at 3 27 54 PM" src="https://github.com/synapsecns/sanguine/assets/104046418/79d54cac-4814-4b8b-bfc7-c2c0343b146f">